### ISSUE TITLE
Announcing DH-keys, missing private part

### ISSUE THREAD

**Issue by staltz** [ssbc/ssb-meta-feeds-dm-spec#1]
Step 1 in this spec says

```
{
  type: 'meta/add/derived',
  purpose: 'chess',
  subfeed: 'ssb:feed/classic/DIoOBMaI1f0mJg+5tUzZ7vgzCeeHh8+zGta4pOjc+k0='
  metadata: {
    encryption: {
      curve: 'curve25519',
      public: 'BtFRgQLgXyq3G48jDrX9OaY/mnPaWnu+kOfx2yEOWwU=' // public DH key
    }
  }
}
```

But we're missing the private part of the keypair.

You [said in this PR](https://github.com/ssbc/ssb-tribes2/pull/34/#discussion_r1023388938) that "_And we are only publishing the public part!!! the private part must be persisted by you_", and I understand that you can't put the private part in this message, but we need the private part to be recoverable from your SSB messages somehow. It's not enough to ask the user to backup `~/.ssb/secret` PLUS the private DM keys for *each* subfeed, it has to be sufficient to only know `~/.ssb/secret`.

And if we are going to announce the private part of the key somewhere on the log, then HOW should we do that? The spec needs to say.

This is important for production, but IMO it's not important for completing our NGI milestone. We need to get stuff shipped but we're constantly creating new tasks for ourselves. Let's call it a day!

I propose doing ed25519-to-curve25519 conversion until we figure this out.

cc @mixmix @arj03

**Comment by arj03** [comment]
I wonder if we can use the same scheme that is used for metafeeds, where if we derive the key deterministically from the seed we store the nonce in there together with the public key. Otherwise it is up to the author to make sure the key is safe in another way.

**Comment by staltz** [comment]
I'd be happy with the nonce-based key derivation idea.

**Comment by mixmix** [comment]
I don't think this needs to be in the main spec. If anything it's an addendum about a pattern which saves you storing heaps of keys. Probably in a different file yes?
